app/main.py

```diff
@@ -1 +1,36 @@
-# write your code here
+class SoftwareEngineer:
+    def __init__(self, name: str):
+        self.name = name
+        self.skills = []
+
+    def learn_skill(self, skill: str):
+        if skill not in self.skills:
+            self.skills.append(skill)
+
+    def __repr__(self):
+        return f"{self.name} has skills: {', '.join(self.skills)}"
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str):
+        super().__init__(name)
+        self.skills.extend(["JavaScript", "CSS", "HTML"])
+
+
+class BackendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str):
+        super().__init__(name)
+        self.skills.extend(["Python", "SQL", "Django"])
+
+
+class FullStackDeveloper(SoftwareEngineer):
+    def __init__(self, name: str):
+        super().__init__(name)
+        # Объединяем навыки Frontend и Backend
+        self.skills.extend(["JavaScript", "CSS", "HTML", "Python", "SQL", "Django"])
```

Comment:
Consider using the `FrontendDeveloper` and `BackendDeveloper` classes to initialize the skills for `FullStackDeveloper`. This would make the code more maintainable and reduce duplication. For example, you could create instances of `FrontendDeveloper` and `BackendDeveloper` and combine their skills.